ganttproject/src/net/sourceforge/ganttproject/GPTreeTableBase.java

```diff
@@ -528,10 +536,97 @@ protected void initTreeTable() {
     doInit();
   }
 
+  private static Comparator<Task> beginComparatorAsc = new Comparator<Task>() {
+    @Override
+    public int compare(Task t1, Task t2) {
+      return t1.getStart().compareTo(t2.getStart());
+    }
+  };
+
+  private static Comparator<Task> beginComparatorDesc = new Comparator<Task>() {
+    @Override
+    public int compare(Task t1, Task t2) {
+      return -t1.getStart().compareTo(t2.getStart());
+    }
+  };
+
+  private static Comparator<Task> endComparatorAsc = new Comparator<Task>() {
+    @Override
+    public int compare(Task t1, Task t2) {
+      return t1.getEnd().compareTo(t2.getEnd());
+    }
+  };
+
+  private Comparator<Task> endComparatorDesc = new Comparator<Task>() {
+    @Override
+    public int compare(Task t1, Task t2) {
+      return -t1.getEnd().compareTo(t2.getEnd());
+    }
+  };
+
+  private class SortTableHeaderRenderer implements TableCellRenderer {
+    private Icon ascIcon;
```

Comment:
Please use `my` prefix: `myAscIcon`

Also, you may want to make these fields final.